sycl/doc/design/IVDEPAndParallelLoopAnnotations.md

```diff
@@ -0,0 +1,338 @@
+   loop-carried memory-ordering dependences. This metadata should not be
+   generated by the SYCL front end in conjunction with ivdep given the
+   difference in their semantics. If a parallel loop feature is added in the
+   future (as descirbed in [the previous section](#marking-parallel-loops)),
+   then this metadata would be generated from that representation.
+4. The built-in `llvm.access.group` metadata should be generated for all memory
+   operations (loads, stores, and function calls) in the body of a loop that has
+   either an ivdep attribute, or is marked as parallel (by the user or the
+   compiler). The access group identifier will be passed to the corresponding
+   loop metadata described above.
+
+## SPIR-V Extension
+
+The SPIR-V extension for this design doc is presented in #13918
+(will update link to PR review is complete). This list
+summarizes the additions:
+
+1. Access groups defined using an LLVM `llvm.access.group` metadata are
+represented as a literal in SPIR-V.
+1. A new instruction `OpAccessGroupListINTEL` is used to aggregate multiple access groups
+   into a single list.
+2. A new Decoration `AccessGroupINTEL` is used to specify which access groups a
+   function call or atomic instruction belongs to.
+3. A new Memory Operand `AccessGroupMaskINTEL` is used to specify which access
+   groups a memory operation belongs to.
+4. Two new Loop Controls are added to represent `llvm.loop.parallel_accesses`
+   and `llvm.loop.no_depends/no_depends_safelen`. They are called
+   `ParallelAccessesINTEL` and `DependencyAccessesINTEL` accordingly.
+   1. `ParallelAccessesINTEL` takes one or more `OpAccessGroupListINTEL` access group
+      lists and specifies that for each list, all the accesses that belong to
+      the specified access groups do not have loop-carried dependences.
+   2. `DependencyAccessesINTEL` takes one or more pairs. The first element in
+      each pair is a list of access groups from `OpAccessGroupListINTEL`, and the second
+      element is an integer `S` that corresponds to the safelen. 
+
+## Translation Rules
+
+### SYCL to LLVM Metadata Translation
+
+   1. For the different versions of ivdep:
+      1. `[[intel::ivdep]]` on a loop should be translated to
+         `llvm.access.group` metadata on all memory instructions and function
+         calls in the loop body and the `llvm.loop.no_depends` metadata should
+         be placed on the loop and passed the same access group(s). The front
+         end is free to assign access groups as it sees fit as long as they all
+         appear in the `llvm.loop.no_depends` metadata.
+      2. `[[intel::ivdep(array)]]` on a loop should be translated to
+         `llvm.access.group` metadata on all memory instructions that use the
+         specified array, and `llvm.loop.no_depends` metadata should be placed
+         on the loop and passed these access group(s). The front end is free to
```

Comment:
Can you make it clear here that we don't expect the FE to tag function calls in this case?